horovod/torch/optimizer.py

```diff
@@ -219,9 +241,13 @@ def synchronize(self):
                     self._allreduce_delay[gp] = self.backward_passes_per_step
                     gp.grad.set_(self._compression.decompress(output, gctx))
             else:
-                output = synchronize(handle)
+                # When handle is a callable function, it returns the aggregated tensor result
+                output = synchronize(handle) if not callable(handle) else handle()
                 self._allreduce_delay[p] = self.backward_passes_per_step
-                p.grad.set_(self._compression.decompress(output, ctx))
+                if p.grad.is_sparse:
+                    p.grad.zero_().add_(self._compression.decompress(output, ctx))
```

Comment:
I see, put sparse result to original dense zero matrix.

Reply:
For some reason you can't `set_` a sparse tensor, so we need to use this hack.  It's kind of ugly.